src/api/spec/controllers/webui/projects/public_key_controller_spec.rb

```diff
@@ -27,7 +27,8 @@
     context 'with a project that has no public key' do
       let(:keyinfo_response) { '<keyinfo />' }
 
-      it { expect(response.status).to eq(404) }
+      it { is_expected.to redirect_to(project_show_path(project)) }
+      it { expect(flash[:error]).not_to be_nil }
```

Comment:
> My thinking is more that, if we care enough to test it, we can also do the extra effort of testing it properly. It's not like this would require hours of development^^.

I disagree. In my experience most codebases start out with the well meaning mentality of "more tests the better" but this usually ends up with tests that test every single detail of the implementation which makes it impossible to refactor without almost completely re-writting the tests. So if you have to re-write the tests you lose the confidence that you have when refactoring with a well thought out test suite. That pretty much makes the tests almost useless apart from when you first write the initial code.

This is basically the problem which has occured with people mocking far too much:
https://www.thoughtworks.com/de/insights/blog/mockists-are-dead-long-live-classicists

Also this video i posted on the previous PR with the similar discussion but I would recommend watching it if you haven't already:
https://www.youtube.com/watch?v=URSWYvyc42M

I think we should strive to write smart tests, not just lots of tests.